src/debug/usersettings-test.js

```diff
@@ -0,0 +1,182 @@
+import chai from 'chai'
+import path from 'path'
+import * as utils from './test-utils'
+chai.should();
+import { ROOT, LANGUAGE_SERVER_ROOT, LANGUAGE_SERVER_WORKSPACE } from './constants'
+import { fs } from 'mz';
+let isDefaultSetting = true;
+let logArray = [];
+
+describe('UserSettings test', () => {
+    let config;
+    let DATA_ROOT;
+    let debugEngine;
+
+    beforeEach(function () {
+        this.timeout(1000 * 20);
+        return (async () => {
+            config = new UserSettings();
+            DATA_ROOT = path.join(ROOT, config.workspaceRoot);
+            debugEngine = await utils.createDebugEngine(DATA_ROOT, LANGUAGE_SERVER_ROOT, LANGUAGE_SERVER_WORKSPACE, config);
+        })();
+    });
+
+    afterEach(() => {
+        isDefaultSetting = false;
+        return debugEngine.close();
+    });
+
+    let debugSetting = function (done) {
+        (async () => {
+            try {
+                await debugEngine.launch();
+                if (config.initialBreakpoints) {
+                    for (let breakpoint of config.initialBreakpoints) {
+                        const breakFile = path.join(DATA_ROOT, config.sourcePath, breakpoint.relativePath);
+                        await debugEngine.setBreakpoints(breakFile, breakpoint.lines);
+                    }
+                }
+                // // starting
+                await debugEngine.startDebug();
+                const terminateEvent = await debugEngine.waitForTerminate();
+                console.log('exiting', terminateEvent);
+                await utils.timeout(1000);
+                logLevelTest();
```

Comment:
logLevelTest is called one time here, how can logArray has two items?

Reply:
I test the loglevel helps through verifying the difference between the first scenario with default set loglevel="INFO" and the second scenario with undefault set loglevel="WARNING". And the debugSetting will be called back twice so it will have two items